sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -50,7 +57,14 @@ public class EventData
         ///   </code>
         /// </example>
         ///
-        public IDictionary<string, object> Properties { get; }
+        public IDictionary<string, object> Properties
+        {
+            get
+            {
+                Interlocked.CompareExchange(ref _properties, new Dictionary<string, object>(), null);
```

Comment:
Not sure if `Interlocked` is overkill here, since we don't normally guarantee thread safety for models.  It seemed like these were important enough that it was a good idea.   An alternative could be something simple like:

```csharp
public IDictionary<string, object> Properties
{
    get => _properties ??= new Dictionary<string, object>();
}
    
public IReadOnlyDictionary<string, object> SystemProperties 
{ 
    get => _systemProperties ??= new Dictionary<string, object>();
}
```

Reply:
IMO it is overkill. It might also confuse future developers about why thread-safety mattered here. Let's use the compound assignment.